core/engine/src/error.rs

```diff
@@ -709,9 +709,37 @@ impl fmt::Debug for JsNativeError {
     }
 }
 
+// FIXME(const-hack): replaced with Box::default once it is `const`
+const fn new_str() -> Box<str> {
+    // SAFETY: miri safe and works in stable
+    unsafe { std::mem::transmute("") }
+}
```

Comment:
Is this guaranteed though? IIRC, `Box<T>` has the same layout as `&T` only for `T: Sized`, but for unsized types they could always change the pointer metadata layout.

If we don't want to rely on this, we can just use `Option<Box<str>>`, which should be const-compatible.

Reply:
Yep, I missed it and miri doesn't flag for it. 
And Since there are plenty of `&'static str` message, can we use `Cow<'static, str>`?

Reply:
Miri's default mode avoids false positives.

This means miri only errors if your program is definitely unsound according to its rules, and avoids warning for some operations that may be unsound, but are not provably unsound.

So "miri said it was okay" is not a proof that your program is sound, it is a proof that you haven't done something _monstrously_ incorrect.

Reply:
Good to know that.

Reply:
Using `Cow` for this sounds good